frontend/components/Nav/index.tsx

```diff
@@ -1,19 +1,51 @@
 import Link from 'next/link';
 import { StyledNav } from './nav.styles';
 
-const Nav: React.FC = () => {
+const Nav: React.FC<{ userData: { permissionLevel: number; userId: string; username: string } | string }> = ({
+    userData
+}) => {
+    const renderNavTabs = () => {
+        if (userData === 'Unauthorised') {
+            return (
+                <Link href='/login'>
+                    <a>Login</a>
+                </Link>
+            );
+        } else {
+            const { permissionLevel, userId, username } = (userData as unknown) as {
+                permissionLevel: number;
+                userId: string;
+                username: string;
+            };
+            const navTabs = [];
+
+            switch (true) {
+                case permissionLevel >= 1:
+                    navTabs.push(
+                        <Link key={1} href='/admin/tutorial-manager'>
+                            <a style={{ marginRight: '1vw' }}>Tutorial Manager</a>
+                        </Link>
+                    );
+                case permissionLevel >= 0:
+                    navTabs.push(
+                        <Link key={0} href='/profile'>
+                            <a>Profile</a>
+                        </Link>
+                    );
+                    break;
+            }
+            return navTabs;
+        }
+    };
```

Comment:
I think we can write a little cleaner code here, with use of useState and useEffect hooks.

Reply:
I think I see where you're going with this. Store current available tabs in state, update them based on user prop? Sounds good.

Reply:
Had a think about this, they're specific switch checks on the permission of the user data, so it just determines an array to render by what the current level is. The current level is passed in by serverSideProps from the page, so I don't think it makes sense to have this stateful as nothing is managed, but rather just caught.